core/src/services/gdrive/writer.rs

```diff
@@ -59,13 +58,7 @@ impl GdriveWriter {
 
         match status {
             StatusCode::OK | StatusCode::CREATED => {
-                let bs = resp.into_body().bytes().await?;
-
-                let file = serde_json::from_slice::<GdriveFile>(&bs)
-                    .map_err(new_json_deserialize_error)?;
-
-                self.file_id = Some(file.id);
-
+                resp.into_body().consume().await?;
```

Comment:
？We need this file_id

Reply:
There is no way to reuse the file id here.

Reply:
A writer which is called multiple times needs this.

Reply:
> A writer which is called multiple times needs this.

GdriveWriter doesn't support muliple write yet which means it will only be called once. Welcome to add it back when we support muliple write.

Reply:
No.Just a normal writer. As gdrive allows files to hold the same name, without reusing the file id, it will be a problem.

Reply:
> As gdrive allows files to hold the same name, without reusing the file id, it will be a problem.

I didn't get your idea. Maybe you can add a test case to cover them?

Reply:
I checked the test case. This situation wasn't covered.

For a one-shot writer, we do the following steps:
```rust
let w = op.writer("path/a");

w.write("hello");
w.write("world");

w.close();
```

Then the end result should be just one file with the content "world".
But gdrive allows files to hold the same name. So if we don't use the file_id which belongs to the file created by the first write step, the result will be two files.

Reply:
Ok. But we can still make up the test.